File: Project/CardValidation.UITest.Android/Pages/MainPage.cs
using CreditCardValidation.UITest;
using CreditCardValidation.UITest.Pages;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Xamarin.UITest.Queries;

namespace CreditCardValidation.UITest.Pages
{
    public class MainPage : AbstractPage, IMainPage
    {
        private Func<AppQuery, AppQuery> CreditCardNumberView { get { return c => c.Id("creditCardNumberText"); } }
        private Func<AppQuery, AppQuery> ValidateButton { get { return c => c.Id("validateButton"); } }

        public MainPage()
        {
            Trait = c => c.Text("Validate Credit Card");
        }

        public void EnterCreditcardNumer(int lengthOfCreditcardNumber)
        {
            Base.App.EnterText(CreditCardNumberView, new string('9', lengthOfCreditcardNumber));
        }

        public void PressValidate()
        {
            Base.App.Tap(ValidateButton);
        }
    }
}

File: Project/features/android/pages/main_page.rb
require 'calabash-android/abase'

class MainPage < Calabash::ABase

  def trait
    "android.widget.TextView text:'Validate Credit Card'"
  end

  def enter_creditcard_number(number)
    enter_text("* marked:'creditCardNumberText'", number)
  end

  def validate
    touch("* marked:'validateButton'")
  end
end
